bears/js/JSONFormatBear.py

```diff
@@ -90,9 +90,15 @@ def run(self, filename, file,
         diff = Diff.from_string_arrays(file, corrected)
 
         if len(diff) > 0:
+            # Appropriate message is displayed to the user based on
+            # json_sort value
+            format_message = ('This file can be reformatted '
+                              'by following indentation.')
+            if json_sort:
+                format_message = ('This file can be reformatted by sorting '
+                                  'keys and following indentation.')
```

Comment:
What if `json_sort` is True, but keys are already sorted? The format message `This file can be reformatted by sorting keys ...` will be confusing.

The next problem is, how do we know if keys are already sorted? A naive approach would be to use `json.dumps` twice, change `sort_keys` value and keep other parameters unchanged, then compare the resulting diff. Any better idea?

Reply:
Yes, that's a problem. Shouldn't we address that in another PR ? Because this PR does pass through the acceptance criteria mentioned in the issue https://github.com/coala/coala-bears/issues/2503.

Reply:
@Makman2 ping :)

Reply:
Those are *minimal* acceptance criteria.

If new additional problems are found in your code, they should be fixed so that your algorithm is stable code.

IMO it is ok to defer determine what type of problem occurred, e.g. by seeing if the keys are unsorted.

But the message needs to be accurate.  Now in light of that, the acceptance criteria are insufficient.  The indentation might be correct, and the keys all sorted, but the dump is different due to setting `escape_unicode`.

For the message to be accurate, it should indicate the values of the three parameters which is using, and be written so that the reader understands that the error could be caused by any of the three parameters being wrong.

Reply:
Yes, I agree. I will work towards a possible solution. Any input towards that would be really helpful.:)